vnext/ReactUWP/Utils/XamlDirectInstance.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <winrt/Windows.UI.Xaml.Core.Direct.h>
+#include <winrt/Windows.UI.Xaml.h>
+
+namespace XD {
+using namespace winrt::Windows::UI::Xaml::Core::Direct;
+}
+
+class XamlDirectInstance {
+ public:
+  static XD::IXamlDirect GetXamlDirect();
+
+ private:
+  static XD::IXamlDirect m_xamlDirect;
```

Comment:
do IXamlDirect objects need to only be used on the UI thread they're created on?  if so this should be thread local for multi window (multi react instance in the same process), scenario we have in production for the Mail app

Reply:
Good point! I haven't made a multi-window app so it is easy to slip from my mind. I am assuming that yes they do need to be used on the UI Thread that created them, but I will check into it.

Reply:
+1 - fixing this should be as simple as declaring this as thread_local

---
In reply to: [303055124](https://github.com/microsoft/react-native-windows/pull/2760#discussion_r303055124) [](ancestors = 303055124)